project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/services/PsiWithBodiesCollectorService.kt

```diff
@@ -0,0 +1,20 @@
+package org.plan.research.minimization.plugin.services
+
+import org.plan.research.minimization.plugin.acceptOnAllKotlinFiles
+import org.plan.research.minimization.plugin.model.PsiWithBodyDDItem
+import org.plan.research.minimization.plugin.psi.BodyElementAcquiringKtVisitor
+
+import com.intellij.openapi.application.readAction
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+
+@Service(Service.Level.PROJECT)
+class PsiWithBodiesCollectorService(private val rootProject: Project) {
```

Comment:
Do we really need a service for this?

Reply:
I don't really see the reasons why we don't. IJ itself has a lot of small services, the one more does not add a big overhead, but for us it opens an opportunity to split the code.

Reply:
We can use just an object here or even a top-level function because there is no "state" or "hard logic". Services are more about a unique place for some stuff with a state.

Reply:
Hmm, yea, some extension function for Project will work find, I agree

Reply:
\+ also you can move here some "find roots" stuff

https://github.com/plan-research/project-minimization/pull/19#discussion_r1819020165